trulens_eval/trulens_eval/feedback.py

```diff
@@ -1,5 +1,353 @@
+  implementation. Here it is a method with the following signature:
+
+    ```python
+    def qs_relevance(self, question: str, statement: str) -> float:
+    ```
+
+    That is, `qs_relevance` is a plain python method that accepts a `question`
+    and a `statement`, both strings, and produces a float (assumed to be between
+    0.0 and 1.0).
+
+- **Feedback constructor** -- The line `Feedback(openai.qs_relevance)` constructs a
+  Feedback object with a feedback implementation.
+
+- **Argument specification** -- The next two lines, `on_input` and `on`, specify how
+  the `qs_relevance` arguments are to be determined from an app record or app
+  definition. The general form of this specification is done using `on` but
+  several shorthands are provided. `on_input` states that the first argument to
+  `qs_relevance` (`question`) is to be the main input of the app. The
+  `on(Select...)` line specifies where the `statement` argument to the
+  implementation comes from. The form of the specification will be discussed in
+  further details in the **Specifying Arguments** section.
+
+- **Aggregation specification** -- The last line `aggregate(numpy.min)` specifies
+  how feedback outputs are to be aggregated. This only applies to cases where
+  the argument specification names more than one value for an input. The second
+  specification, for `statement` was of this type. The input to `aggregate` must
+  be a method which can be imported globally. This requirement is further
+  elaborated in the next section. This function is called on the `float` results
+  of feedback function evaluations to produce a single float. The default is
+  `numpy.mean`.
+
+The result of these lines is that `f_qs_relevance` can be now be run on
+app/records and will automatically select the specified components of those
+apps/records:
+
+```python
+record: Record = ...
+app: App = ...
+
+feedback_result: FeedbackResult = f_qs_relevance.run(app=app, record=record)
+```
+
+The object can also be provided to an app wrapper for automatic evaluation:
+
+```python
+app: App = tru.Chain(...., feedbacks=[f_qs_relevance])
+```
+
+## Specifying Implementation Function and Aggregate
+
+The function or method provided to the `Feedback` constructor is the
+implementation of the feedback function which does the actual work of producing
+a float indicating some quantity of interest. Any callable can be provided here
+but there are additional requirements if your app uses the "deferred" feedback
+evaluation mode (when `feedback_mode=FeedbackMode.DEFERRED` are specified to app
+constructor). In those cases the callables must be methods that are globally
+importable (see the next section for details). The function/method performing
+the aggregation has the same requirements.
+
+The mapping of record/app contents to the arguments to the specified feedback
+implementation is provided via the `on_...` methods. These are discussed in a
+later section.
```

Comment:
lets just remove this; things are a bit verbose; and they will get to it if they are reading this far

Reply:
done